### DISCUSSION TITLE
Possible issue in  /grbl/gcode.c on STM32CubeIDE 1.15.0

### DISCUSSION THREAD

**Discussion by samitray** [grblHAL/core#492] [General]
Hi,
Recently updated STM32Cube from 1.12.0 to 1.15.0 and saw  some error/warning messages
1. In   /grbl/gcode.c    function gc_execute_block (char *block) on line 3537 there is warning : pointer used after 'free' [-Wuse-after-free] 
    i think line 3535  output_command_t *next = plan_data.output_commands;             should be 
                               output_command_t *next = plan_data.output_commands->next;
2. Only for STM32F1XX branch Error in /Src/sysmem.c   function  _sbrk(int incr)   c_addr_t be replaced by void*
3. In linker files there are warnings, the section below need to have readonly tag
    .ARM.extab (READONLY) :
    .ARM (READONLY) :
    .preinit_array (READONLY) :
    .init_array (READONLY) :
    .fini_array (READONLY) :

Haven't tested out these changes on a physical machine... hope to test in the next few days and report if there are any misbehavior
due to these changes

**Comment by terjeio** [comment]
1. Fixed in the [latest commit](https://github.com/grblHAL/core/commit/e192ec566b4259589d05d3c2ce6c25210fcd2743), luckily there was a low risk for this code to ever beeing run.
2. Fixed late last year, you have an old version?
3. I'll have to check if this fix messes up the Web Builder, then update all the loader files. I have not detected any issues related to it.

**Comment by samitray** [comment]
1. yes that branch has low probability
2. I pulled the July 2023 tagged release for STM32F1xx two days back, think that (void *)  is only in the july 2023 tag, other branches have the fix. Pulled july 2023 release because recent release are not able to fit into stm32F103
3. Ok.... will notify if there are any "weird" problems

**Reply by terjeio** [reply]
> Pulled july 2023 release because recent release are not able to fit into stm32F103

FYI release builds built with the CubeIDE fits with some room to spare, debug builds does not.

**Reply by terjeio** [reply]
3. Adding (READONLY) to the listed sections breaks the Web Builder so I will defer updating the linker scripts until the backend can be updated.

**Comment by samitray** [comment]
ok..
